samza-core/src/main/java/org/apache/samza/coordinator/CoordinationUtils.java

```diff
@@ -41,4 +41,6 @@
   LeaderElector getLeaderElector(); // leaderElector is unique based on the groupId
 
   Latch getLatch(int size, String latchId);
+
+  Lock getLock();
```

Comment:
I'd rather named it CoordinationLock (or something), to reduce confusion between this and java concurrency lock.